LOPS-741 - PantheonSecrets Documentation (#8975)
pantheon-systems/documentation: source/content/guides/secrets/01-introduction.md

---
title: Pantheon Secrets Guide
subtitle: Introduction
description: Securely store secrets in the Pantheon Platform.
type: terminuspage
contributors: [stovak]
contenttype: [guide]
innav: [true]
categories: [secrets]
cms: [drupal, wordpress]
audience: [development]
product: [secrets]
integration: [--]
tags: [reference, cli, local, terminus, workflow]
permalink: docs/guides/secrets
reviewed: "2024-05-01"
---

Pantheon Secrets is key to maintaining industry best practices for secure builds and application implementation. This feature provides a convenient mechanism for you to manage your secrets and API keys directly on the Pantheon platform.
This guide covers features and use cases of the Pantheon Secrets feature; it could also be referred as Secrets Manager because that is the Terminus plugin name.
Features

* Secure: secrets are encrypted at rest.
* Easy to use: create and update secrets via Terminus.

> * Governability: secrets could be set at organization level and shared to all the sites owned by that organization.

Reviewer: "Governability" doesn't align with the other lines. "Governable" would be consistent.

* Overridable: secrets could be overridden at environment level when needed.
  

Support Community submitted bugs and feature requests can be found in the plugin, SDK or Drupal module issue queues. You can create an issue in the corresponding component if you don't see your bug or feedback listed.

[Contact Support](https://dashboard.pantheon.io/#support/support/all) if you have questions or need help with Terminus.

## Contribute to Pantheon Secrets

> The plugin, SDK or Drupal module are open source. You can view the projects and contribute, file issues and submit feature requests.

Author: https://github.com/pantheon-systems/documentation/pull/8975/commits/9fe9332282daecd2aaedf3a93f968c33d68b63bb

---
title: Pantheon Secrets Guide
subtitle: Introduction
description: Securely store secrets in the Pantheon Platform.
contributors: [stovak]
contenttype: [guide]
innav: [true]
categories: [secrets]
cms: [drupal, wordpress]
audience: [development]
product: [secrets]
integration: [--]
tags: [reference, cli, local, terminus, workflow]
permalink: docs/guides/secrets
reviewed: "2024-07-30"
showtoc: true
---

Pantheon Secrets is key to maintaining industry best practices for secure builds and application implementation. This feature provides a convenient mechanism for you to manage your secrets and API keys directly on the Pantheon platform.

This guide covers features and use cases of the Pantheon Secrets feature; it could also be referred as Secrets Manager because that is the Terminus plugin name.

> ## Features

Reviewer 2: The feature list here should be aligned with what's specified in the plugin's README file

---
title: Pantheon Secrets Guide
subtitle: Introduction
description: Securely store secrets in the Pantheon Platform.
type: terminuspage
contributors: [stovak]
contenttype: [guide]
innav: [true]
categories: [secrets]
cms: [drupal, wordpress]
audience: [development]
product: [secrets]
integration: [--]
tags: [reference, cli, local, terminus, workflow]
permalink: docs/guides/secrets
reviewed: "2024-05-01"
---

> Pantheon Secrets is key to maintaining industry best practices for secure builds and application implementation. This feature provides a convenient mechanism for you to manage your secrets and API keys directly on the Pantheon platform.

Author: I don't care about the langouage or the language, either.

This guide covers features and use cases of the Pantheon Secrets feature; it could also be referred as Secrets Manager because that is the Terminus plugin name.

Features:

* **Secure**: secrets are encrypted at rest.
* **Easy to use**: create and update secrets via Terminus.
* **Governable**: secrets could be set at organization level and shared to all the sites owned by that organization.
* **Overridable**: secrets could be overridden at environment level when needed.
  

> Support Community-submitted bugs and feature requests can be found in the plugin, SDK or Drupal module issue queues. You can create an issue in the corresponding component if you don't see your bug or feedback listed.

Reviewer 3: I was about to start linking the words, but we're linking the exact same things we link in "contributing", which makes me think maybe they should be combined....

> ### Support
> Community-submitted bugs and feature requests can be found in their respective repositories on GitHub.
> 
> * [Terminus Secrets Manager Plugin](https://github.com/pantheon-systems/terminus-secrets-manager-plugin)
> * [Secrets SDK](https://github.com/pantheon-systems/customer-secrets-php-sdk)
> * [pantheon_secrets (Drupal module)](https://github.com/pantheon-systems/pantheon_secrets)
> 
> [Contact Support](https://dashboard.pantheon.io/#support/support/all) if you have questions or need help with Terminus.

[Contact Support](https://dashboard.pantheon.io/#support/support/all) if you have questions or need help with Terminus.

## Disambiguation

> Previous to the release of the [SECRETS MANAGER PLUGIN](https://github.com/pantheon-systems/terminus-secrets-manager-plugin), Pantheon offered the [SECRETS PLUGIN](https://github.com/pantheon-systems/terminus-secrets-plugin). The SECRETS PLUGIN and the method for storing secrets it uses is deprecated and should not be used.

Reviewer 3: Rewritten.  No caps.  

> ### Disambiguation
> 
> The [Terminus Secrets Manager Plugin](https://github.com/pantheon-systems/terminus-secrets-manager-plugin) should not be confused with the older [terminus-secrets-plugin](https://github.com/pantheon-systems/terminus-secrets-plugin).  The key differences are:
> 
> - The Terminus Secrets Manager Plugin stores secrets in an encrypted backend service.
> - The older secrets plugin simply stores them in a /files/private/secrets.json file.
> 
> Once the Pantheon Secrets service becomes generally available, the older terminus-secrets-plugin will be discontinued.

## Contribute to Pantheon Secrets

The plugin, SDK or Drupal module are open source. You can view the projects and contribute, file issues and submit feature requests.

> * [SECRETS MANAGER PLUGIN](https://github.com/pantheon-systems/terminus-secrets-manager-plugin)

Reviewer 3: Nocaps.

This guide covers features and use cases of the Pantheon Secrets feature; it could also be referred as Secrets Manager because that is the Terminus plugin name.

### Features:

* **Secure**: secrets are encrypted at rest.
* **Easy to use**: create and update secrets via Terminus.
* **Governable**: secrets could be set at organization level and shared to all the sites owned by that organization.
* **Overridable**: secrets could be overridden at environment level when needed.

### Support
Community-submitted bugs and feature requests can be found in their respective repositories on GitHub.

* [Terminus Secrets Manager Plugin](https://github.com/pantheon-systems/terminus-secrets-manager-plugin)
* [Secrets SDK](https://github.com/pantheon-systems/customer-secrets-php-sdk)
* [pantheon_secrets (Drupal module)](https://github.com/pantheon-systems/pantheon_secrets)

[Contact Support](https://dashboard.pantheon.io/#support/support/all) if you have questions or need help with Terminus.

### Disambiguation

> The [Terminus Secrets Manager Plugin](https://github.com/pantheon-systems/terminus-secrets-manager-plugin) should not be confused with the older [terminus-secrets-plugin](https://github.com/pantheon-systems/terminus-secrets-plugin).  The key differences are:

Reviewer: s/should not be confused with/replaces/

> The [Terminus Secrets Manager Plugin](https://github.com/pantheon-systems/terminus-secrets-manager-plugin) should not be confused with the older [terminus-secrets-plugin](https://github.com/pantheon-systems/terminus-secrets-plugin).  The key differences are:

Reviewer: ```suggestion
The [Terminus Secrets Manager Plugin](https://github.com/pantheon-systems/terminus-secrets-manager-plugin) replaces the older [terminus-secrets-plugin](https://github.com/pantheon-systems/terminus-secrets-plugin).  The key differences are:
```

Reviewer 3: Made this change.

- The Terminus Secrets Manager Plugin stores secrets in an encrypted backend service.

> - The older secrets plugin simply stores them in a /files/private/secrets.json file.

Reviewer: ```suggestion
- The older secrets plugin simply writes unencrypted values in a json file in `/files/private`.
```
